### DISCUSSION TITLE
I'm working on converting IFC to CITYJSON.

### DISCUSSION THREAD

**Discussion by Pdongjaelee** [cityjson/cjio#186] [General]
hello.
I'm working on converting IFC to CITYJSON.

Currently, IFCOPENSHELL is used to extract geometry information from IFC data.

Now I just need to create this data as a CITYJSON file, but I keep getting an error.

I mainly used cjio to create cityjson, but is there another way?

**Comment by hugoledoux** [comment]
this is a discussion, not an issue wiht the specs

**Comment by Pdongjaelee** [comment]
hello.
I am using cjio to convert ifc to cityjson.

Could you please take a look at my code?

**Comment by hugoledoux** [comment]
cjio cannot convert IFC to CityJSON I'm afraid, it only reads CityJSON and OBJ as input.

**Comment by Pdongjaelee** [comment]
Ah, I have a lot of obj files.

If I have an obj file, can I convert it to cityjson using cjio?

**Reply by hugoledoux** [reply]
Sorry OBJ cannot but an [OFF file](https://en.wikipedia.org/wiki/OFF_(file_format)) is possible to read as input, it gets converted to a "GenericCityObject".

You can convert OBJ to OFF easily with MeshLab.

`cjio cube.off info`

**Reply by Pdongjaelee** [reply]
Oh, then there is currently no way to convert ifc, obj to cityjson using cjio?

**Reply by hugoledoux** [reply]
No. cjio is designed to manipulate CityJSON files and export to a few easy formats like glTF or CityJSONSeq.

There are papers to convert IFC to CityJSON, some have code, but it's not a simple operation.

**Reply by Pdongjaelee** [reply]
Ah, I am currently using the ifc library to extract the parts needed for cityjson.
Just extract it into cityjson.
I was using cjio for that.

Has there ever been a situation like this?

**Reply by jaspervdv** [reply]
We are working on an application that abstracts IFC models and stores it as a CityJSON model. It does not extract the ifc data and store it in the CityJSON format 1:1 but it might still be interesting for you to take a look at. 

https://github.com/jaspervdv/IFC_BuildingEnvExtractor

**Reply by Pdongjaelee** [reply]
hello.
I successfully converted cityjson to lod3.2 yesterday.

But there is a problem.
he external and internal forms of this converted cityjson have been converted intact.
However, there is no representation of converted objects in ifc.
Objects like ifcbeam.

**Comment by Pdongjaelee** [comment]
https://github.com/jaspervdv/IFC_BuildingEnvExtractor
Oh, so using this link, is it possible to abstract the IFC model and save it as a CityJSON model?

That's what it looks like when I take a quick look at the link.
